src/libasr/codegen/wasm_to_x86.cpp

```diff
@@ -154,6 +154,41 @@ class X86Visitor : public WASMDecoder<X86Visitor>,
         m_a.asm_push_r32(X86Reg::eax);
     }
 
+    void handle_I32Compare(const std::string &compare_op) {
+        unique_id.push_back(std::to_string(offset));
```

Comment:
@Thirumalai-Shaktivel, please, could you possibly share why we need to store the `offset` in `unique_id`? Are we using it later outside the `handleI32Compare()` function?

If it is not used later, we need/should not store it, I think.

Reply:
Yup, I forgot to pop the uniqu_id!
Now, If I think of ti, we have options:
1. 
```diff
diff --git a/src/libasr/codegen/wasm_to_x86.cpp b/src/libasr/codegen/wasm_to_x86.cpp
index cf968d5c4..12ea3b156 100644
--- a/src/libasr/codegen/wasm_to_x86.cpp
+++ b/src/libasr/codegen/wasm_to_x86.cpp
@@ -199,30 +199,29 @@ class X86Visitor : public WASMDecoder<X86Visitor>,
     }
 
     void handle_I32Compare(const std::string &compare_op) {
-        unique_id.push_back(std::to_string(offset));
         m_a.asm_pop_r32(X86Reg::ebx);
         m_a.asm_pop_r32(X86Reg::eax);
         m_a.asm_cmp_r32_r32(X86Reg::eax, X86Reg::ebx);
         if (compare_op == "Eq") {
-            m_a.asm_je_label(".compare_1" + unique_id.back());
+            m_a.asm_je_label(".compare_1" + std::to_string(offset));
...
```
2.
```diff
diff --git a/src/libasr/codegen/wasm_to_x86.cpp b/src/libasr/codegen/wasm_to_x86.cpp
index cf968d5c4..2e23d7230 100644
--- a/src/libasr/codegen/wasm_to_x86.cpp
+++ b/src/libasr/codegen/wasm_to_x86.cpp
@@ -224,6 +224,7 @@ class X86Visitor : public WASMDecoder<X86Visitor>,
         m_a.asm_mov_r32_imm32(X86Reg::eax, 1);
         m_a.add_label(".compare.end_" + unique_id.back());
         m_a.asm_push_r32(X86Reg::eax);
+        unique_id.pop_back();
     }
 
     void visit_I32Eq() { handle_I32Compare("Eq"); }
```
which do you prefer the most?

Reply:
As you said, yup, it is not used later on. So, Instead of storing the unique_id, we just use `std::to_string(offset)`. I think option 1 is good to go.

Reply:
Also, the reason why I used the unique_id is because: I thought `offset` value would be incremented inside `handle_compareI32()` function. I checked and it seems the offset is not modified. So, we can use `std::to_string(offset)`.
I will make the able change with the loop stmt.
Thanks for noticing it.

Reply:
>So, Instead of storing the unique_id, we can just use std::to_string(offset)

Yup, we can just use `offset` as a label.

Reply:
I incorporated this change into this commit [WASM_X86: Use offset directly as label](https://github.com/lcompilers/lpython/pull/1285/commits/068563a24a95311c321a99b10fbf5f95f2a76a31)